pages/admin/dashboard.js

```diff
@@ -0,0 +1,229 @@
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import { CircularProgress, Typography } from "@material-ui/core";
+import PropTypes, { array } from "prop-types";
+import styles from "../../styles/GameRow.module.css";
+import game from "../../helpers/game";
+
+import firebase from "../../utils/firebase";
+import AdminGameView from "./game";
+
+import Link from "next/link";
+import {
+  Box,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Container,
+  IconButton,
+  Paper,
+  Snackbar,
+  TextField,
+} from "@material-ui/core";
+
+export default function Dashboard() {
+  const [gameArrayState, setGameArrayState] = useState([]);
+  const [gameClick, setGameClick] = useState(false);
+  const [login, setLogin] = useState(false);
+  const [gamePin, setGamePin] = useState();
+
+  const router = useRouter();
+
+  if (firebase.apps.length === 0) {
+    console.log("not initialized");
+  } else {
+    firebase.auth().onAuthStateChanged(function (user) {
+      if (user) {
+        setLogin(true);
+      } else {
+        //no user logged in, so go back to admin
+        setLogin(false);
+        router.replace(`/admin?login=signOut`);
+      }
+    });
+  }
+
+  useEffect(async () => {
+    var data;
+    const snapshot = await firebase
+      .database()
+      .ref(`games`)
+      .on("value", (subsnapshot) => {
+        data = subsnapshot.val();
+        //console.log(data);
+        setGameArrayState(data);
+      });
+  }, []);
```

Comment:
unsubscribe from the firebase listener. this will prevent you from getting errors about updating state after the component is unmounted

[example](https://github.com/garyhtou/kahoot-buzzfeed/blob/38c559cb35a842c1e59483ff6b48af378874dac1/components/gameQuestion.js#L45-L63)